### PR TITLE
Update sentieon dnascope

### PR DESCRIPTION
**Pull request by asp8200**
Updating Sentieon Dnascope module to facilitate both vcf and gvcf output from one invocation.

Added option for `pcr_indel_model`.

Added some extra pytests.

## PR checklist

Closes #XXX 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

**Review by ramprasadn — APPROVED** [review]
LGTM! 👍🏻

File: modules/nf-core/sentieon/dnascope/main.nf [hunk 1]
```diff
@@ -29,14 +33,27 @@ process SENTIEON_DNASCOPE {
...
-    def args2     = task.ext.args2  ?: ''
-    def interval  = call_interval   ? "--interval ${call_interval}" : ''
-    def dbsnp_str = dbsnp           ? "-d ${dbsnp}"                 : ''
-    def model     = ml_model        ? "--model ${ml_model}"         : ''
-    def prefix    = task.ext.prefix ?: "${meta.id}"
+    def args                      = task.ext.args                      ?: ''  // options for the driver
+    def args2                     = task.ext.args2                     ?: ''  // options for the vcf generation
+    def args3                     = task.ext.args3                     ?: ''  // options for the gvcf generation
+    def interval                  = intervals                          ? "--interval ${intervals}"               : ''
+    def dbsnp_str                 = dbsnp                              ? "-d ${dbsnp}"                           : ''
```
Comment (ramprasadn) [inline modules/nf-core/sentieon/dnascope/main.nf:40]:
```suggestion
    def dbsnp_cmd                 = dbsnp                              ? "-d ${dbsnp}"                           : ''
```

File: modules/nf-core/sentieon/dnascope/main.nf [hunk 2]
```diff
@@ -29,14 +33,27 @@ process SENTIEON_DNASCOPE {
...
+    def model_cmd                 = ml_model                           ? " --model ${ml_model}"                  : ''
+    def pcr_indel_model_cmd       = pcr_indel_model                    ? " --pcr_indel_model ${pcr_indel_model}" : ''
+    def prefix                    = task.ext.prefix                    ?: "${meta.id}"
     def sentieon_auth_mech_base64 = task.ext.sentieon_auth_mech_base64 ?: ''
     def sentieon_auth_data_base64 = task.ext.sentieon_auth_data_base64 ?: ''
+    def vcf_cmd                   = ""
+    def gvcf_cmd                  = ""
+    def base_cmd                  = '--algo DNAscope ' + dbsnp_str
```
Comment (asp8200) [inline modules/nf-core/sentieon/dnascope/main.nf:48]:
```suggestion
    def base_cmd                  = '--algo DNAscope ' + dbsnp_cmd
```

**Review by maxulysse — APPROVED** [review]
Toppen

**Pull request merged**